enterprise/server/auth/auth.go

```diff
@@ -349,74 +362,80 @@ func (a *OpenIDAuthenticator) lookupUserFromSubID(subID string) (*tables.User, e
 	return user, err
 }
 
-func (a *OpenIDAuthenticator) lookupGroupFromAPIKey(apiKey string) (*tables.Group, error) {
+func (a *OpenIDAuthenticator) lookupAPIKeyGroupFromAPIKey(apiKey string) (*apiKeyGroup, error) {
 	dbHandle := a.env.GetDBHandle()
 	if dbHandle == nil {
 		return nil, status.FailedPreconditionErrorf("No handle to query database")
 	}
-	tg := &tables.Group{}
+	akg := &apiKeyGroup{}
 	err := dbHandle.TransactionWithOptions(db.StaleReadOptions(), func(tx *gorm.DB) error {
 		existingRow := tx.Raw(`
-			SELECT g.* FROM `+"`Groups`"+` AS g, APIKeys AS ak
+			SELECT ak.value AS key, ak.capabilities, g.group_id
```

Comment:
does this change the performance of this query at all?

Reply:
We should benchmark these against mysql as well as that's what we run in prod. Can you double check this and the one below are both good there too? (This is easy to setup locally on a Linux box, lmk if you want help)

Reply:
Oof, good thing you mentioned that, because there was a syntax error in SQL that doesn't happen in sqlite (it doesn't let allow `key` as a field name, so I wrapped it in backticks to make it happy). It thinks "Groups" is special too but the query works without backticking "Groups." Anyway, backticked both.

So, here's the query plan for this query on mysql 5.7 (it's the same before and after):

![image](https://user-images.githubusercontent.com/2414826/107311701-cd5ea700-6a5c-11eb-87ce-4474eab6714f.png)

And here's the plan for the one below:

![image](https://user-images.githubusercontent.com/2414826/107312591-c173e480-6a5e-11eb-9b37-c2a160a97e92.png)

Before adding the join, this was the plan:

![image](https://user-images.githubusercontent.com/2414826/107312646-dea8b300-6a5e-11eb-8820-6533f47b8b0d.png)